src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,268 @@
+                    newY = checkIfPlaceable(x1,y,i,world);
+                    final BlockPos row3 = new BlockPos(x1, newY, i);
+                    world.setBlockState(row3, ConstructionTape.withProperty(FACING, EnumFacing.EAST));
+                    newY = checkIfPlaceable(x3,y,i,world);
+                    final BlockPos row4 = new BlockPos(x3, newY, i);
+                    world.setBlockState(row4, ConstructionTape.withProperty(FACING, EnumFacing.WEST));
+                }
+            }
+            newY = checkIfPlaceable(x1,y,z1,world);
+            final BlockPos corner1 = new BlockPos(x1, newY, z1);
+            newY = checkIfPlaceable(x1,y,z3,world);
+            final BlockPos corner2 = new BlockPos(x1, newY, z3);
+            newY = checkIfPlaceable(x3,y,z1,world);
+            final BlockPos corner3 = new BlockPos(x3, newY, z1);
+            newY = checkIfPlaceable(x3,y,z3,world);
+            final BlockPos corner4 = new BlockPos(x3, newY, z3);
+            world.setBlockState(corner1, ConstructionTapeCorner.withProperty(FACING, EnumFacing.SOUTH));
+            world.setBlockState(corner2, ConstructionTapeCorner.withProperty(FACING, EnumFacing.EAST));
+            world.setBlockState(corner3, ConstructionTapeCorner.withProperty(FACING, EnumFacing.WEST));
+            world.setBlockState(corner4, ConstructionTapeCorner.withProperty(FACING, EnumFacing.NORTH));
+        }
+
+    /**
+     ** Proxy to remove the tape also with the building only.
+     * @param building the building.
+     * @param world the world.
+     */
+    public static void removeConstructionTape(@NotNull AbstractBuilding building, @NotNull World world)
+    {
+        removeConstructionTape(new WorkOrderBuild(building, 1), world);
+    }
+
+    public static void removeConstructionTape(@NotNull WorkOrderBuild workOrder,@NotNull World world)
+    {
+        final StructureWrapper wrapper = new StructureWrapper(world, workOrder.getStructureName());
+        final BlockPos pos = workOrder.getBuildingLocation();
+        int tempRotation = 0;
+        final IBlockState Air = Blocks.AIR.getDefaultState();
+        if (workOrder.getRotation() == 0 && !(workOrder instanceof WorkOrderBuildDecoration))
+        {
+            final IBlockState blockState = world.getBlockState(pos);
+            if (blockState.getBlock() instanceof AbstractBlockHut)
+            {
+                tempRotation = BlockUtils.getRotationFromFacing(blockState.getValue(AbstractBlockHut.FACING));
+            }
+        }
+        else
+        {
+            tempRotation = workOrder.getRotation();
+        }
+        wrapper.rotate(tempRotation, world, workOrder.getBuildingLocation());
+        wrapper.setPosition(pos);
+        int x1 = wrapper.getPosition().getX() - wrapper.getOffset().getX() - 1;
+        int z1 = wrapper.getPosition().getZ() - wrapper.getOffset().getZ() - 1;
+        int x3 = wrapper.getPosition().getX() + (wrapper.getWidth() - wrapper.getOffset().getX());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'x3' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)